front/lib/resources/content_fragment_resource.ts

```diff
@@ -0,0 +1,82 @@
+import type { ContentFragmentType, Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type {
+  Attributes,
+  CreationAttributes,
+  ModelStatic,
+  Transaction,
+} from "sequelize";
+
+import type { Message } from "@app/lib/models";
+import { BaseResource } from "@app/lib/resources/base_resource";
+import { ContentFragmentModel } from "@app/lib/resources/storage/models/content_fragment";
+import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ContentFragmentResource
+  extends ReadonlyAttributesType<ContentFragmentModel> {}
+export class ContentFragmentResource extends BaseResource<ContentFragmentModel> {
+  static model: ModelStatic<ContentFragmentModel> = ContentFragmentModel;
+
+  // TODO(2024-02-20 flav): Delete Model from the constructor, once `update` has been migrated.
+  constructor(
+    model: ModelStatic<ContentFragmentModel>,
+    blob: Attributes<ContentFragmentModel>
+  ) {
+    super(ContentFragmentModel, blob);
+  }
+
+  static async makeNew(
+    blob: CreationAttributes<ContentFragmentModel>,
+    transaction?: Transaction
+  ) {
+    const contentFragment = await ContentFragmentModel.create(
+      {
+        ...blob,
+      },
+      {
+        transaction,
+      }
+    );
+
+    return new this(ContentFragmentModel, contentFragment.get());
+  }
+
+  async delete(transaction?: Transaction): Promise<Result<undefined, Error>> {
+    try {
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+        transaction,
+      });
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  renderFromMessage({ message }: { message: Message }): ContentFragmentType {
```

Comment:
We could probably flatten the parameters here.
```suggestion
  renderFromMessage(message: Message): ContentFragmentType {
```

Reply:
We generally try to unflatten as much as possible to avoid the future addition of more argument unflatten and then errors due to type uncertainties

Reply:
I agree with this pattern, except when the method is very specific 🙃.